ipaserver/install/ipa_server_certinstall.py

```
@@ -157,6 +157,30 @@ def install_http_cert(self):
         os.chown(os.path.join(dirname, 'key3.db'), 0, pent.pw_gid)
         os.chown(os.path.join(dirname, 'secmod.db'), 0, pent.pw_gid)
 
+    def check_chain(self, pkcs12_filename, pkcs12_pin, nssdb):
+        # create a temp nssdb
+        with NSSDatabase() as tempnssdb:
+            db_password = ipautil.ipa_generate_password()
+            db_pwdfile = ipautil.write_tmp_file(db_password)
+            tempnssdb.create_db(db_pwdfile.name)
+
+            # import all the CA certs from nssdb into the temp db
+            for nickname, flags in nssdb.list_certs():
+                if 'C' in flags:
+                    cert = nssdb.get_cert_from_db(nickname)
+                    tempnssdb.add_cert(cert, nickname, flags)
+
+            # now import the server cert and check its validity
+            cert, nickname, flags = installutils.get_server_cert_from_pkcs12(
+                    pkcs12_filename, pkcs12_pin)
+            tempnssdb.add_cert(cert, nickname, flags)
+            try:
+                tempnssdb.verify_server_cert_validity(
+                    nickname, api.env.host)
+            except ValueError as e:
+                return False
```

Rather than returning a value, raise a `ScriptError` here.